dask/dataframe/core.py

```diff
@@ -7694,13 +8068,13 @@ def has_parallel_type(x):
     return get_parallel_type(x) is not Scalar
 
 
-def new_dd_object(dsk, name, meta, divisions, parent_meta=None):
+def new_dd_object(dsk, name, meta, divisions, parent_meta=None, **metdata_kwargs):
```

Comment:
This looks like a great opportunity for consolidation of kwargs into a single metadata object (I suspect that's on your mind here, but out-of-scope for this POC):

```python
def new_dd_object(dsk, name, partition_metadata):
```